supertokens_python/recipe/dashboard/api/signin.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+#
+# This software is licensed under the Apache License, Version 2.0 (the
+# "License") as published by the Apache Software Foundation.
+#
+# You may not use this file except in compliance with the License. You may
+# obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from supertokens_python.recipe.dashboard.interfaces import APIInterface, APIOptions
+
+from supertokens_python.exceptions import raise_bad_input_exception
+from supertokens_python.normalised_url_path import NormalisedURLPath
+from supertokens_python.querier import Querier
+from supertokens_python.utils import send_200_response
+
+
+# pylint: disable=unused-argument
+async def handle_sign_in_api(api_implementation: APIInterface, api_options: APIOptions):
+    body = await api_options.request.form_data()
+
+    if not body["email"]:
```

Comment:
See how we do it in `handle_user_email_verify_put` 

Basically, you should assign vars like `email = body.get("email")`

and then in the condition, use `if email is None or not isinstance(email, str):`

`not var` is too broad and should be avoided.